cypress/tests/bucket-policy-spec.ts

```diff
@@ -0,0 +1,92 @@
+import { DATA_FEDERATION_NAMESPACE, SECOND, MINUTE } from '../constants/common';
+import { dataSourceNSFS } from '../mocks/data-source';
+import { MCGMSCommon, projectNameSpace } from '../views/common';
+import { pvc } from '../views/pvc';
+
+describe('Bucket policy page', () => {
+  const singleBucketPolicy = 'e2e-bucket-single';
+  const pvcName = 'e2e-pvc';
+  const dataSourceName = 'e2e-nsfs-data-source';
+
+  before(() => {
+    cy.login();
+    cy.clickNavLink(['Storage', 'PersistentVolumeClaims']);
+    projectNameSpace.selectOrCreateProject(DATA_FEDERATION_NAMESPACE);
+    pvc.createPVC(pvcName, '1', 'gp2', false);
+  });
+
+  beforeEach(() => {
+    MCGMSCommon.visitBucketPolicyList();
+  });
+
+  after(() => {
+    /**
+     * Need some time for BucketClass to get cleaned-up properly before deleting NamespaceStore,
+     * else we get an error from server: admission webhook "admissionwebhook.noobaa.io" denied the request:
+     * cannot complete because nsr "e2e-nsfs-data-source" in "IN_USE" state.
+     * Even though BucketClass is actually deleted, there is some deplay for it to get reflected for NamespaceStore.
+     */
+    cy.wait(30 * SECOND);
+    cy.exec(
+      `oc delete namespacestores ${dataSourceName} -n ${DATA_FEDERATION_NAMESPACE}`,
+      { failOnNonZeroExit: false }
+    ).then(() => {
+      cy.exec(`oc delete pvc ${pvcName} -n ${DATA_FEDERATION_NAMESPACE}`, {
+        timeout: 3 * MINUTE,
+        failOnNonZeroExit: false,
+      });
+    });
+
+    cy.logout();
+  });
+
+  // ToDo(Sanjal): Need to refactor and add more test blocks for "Multi" and "Cache" types as well.
+  // right now we are only creating "nsfs" (FileSystem) type NamespaceStore which is only allowed with BucketClass of type "Single".
+  it('creates Bucket policy with single data source', () => {
+    cy.exec(
+      `echo '${JSON.stringify(
+        dataSourceNSFS(dataSourceName, pvcName, 'e2e-subPath')
+      )}' | oc create -f -`
+    ).then(() => {
+      cy.byTestID('item-create').click();
+      cy.byTestID('bucket-name-text').type(singleBucketPolicy);
+      cy.byTestID('read-write-dropdown')
```

Comment:
![image](https://user-images.githubusercontent.com/22158580/179449904-207890c5-6063-44f6-b4af-0fc981bf1577.png)

failing

Reply:
did u add the "data-id" ??

Reply:
where and what I need to add? I just pulled your branch and ran the tests

Reply:
CI and local both are passing for me...

Reply:
let's wait for CI once again...